alica_engine/include/engine/AlicaTimer.h

```diff
@@ -0,0 +1,114 @@
+#pragma once
+
+#include "engine/AlicaClock.h"
+#include "engine/IAlicaTimer.h"
+
+#include <atomic>
+#include <condition_variable>
+#include <functional>
+#include <memory>
+#include <stdexcept>
+#include <thread>
+
+namespace alica
+{
+
+class AlicaSystemTimerImpl
+{
+public:
+    using TimerCb = std::function<void()>;
+
+    AlicaSystemTimerImpl(TimerCb&& userCb, alica::AlicaTime period)
+            : _userCb(std::move(userCb))
+            , _period(period.inMilliseconds())
+            , _isActive(false)
+    {
+    }
+
+    ~AlicaSystemTimerImpl()
+    {
+        if (_isActive) {
+            stop();
+        }
+    }
+
+    void start()
```

Comment:
I think implementation doesn't  need to be separate here, but even if it is I think function definitions should be in cpp file

Reply:
take a look, see if it's fine now